lib/puppet_litmus/serverspec.rb

```diff
@@ -171,16 +171,21 @@ def bolt_upload_file(source, destination, opts = {}, options = {})
     result
   end
 
+  # rubocop:disable Layout/TrailingWhitespace
+
   # Runs a task against the target system.
   #
   # @param task_name [String] The name of the task to run.
   # @param params [Hash] key : value pairs to be passed to the task.
-  # @param opts [Hash] Alters the behaviour of the command. Valid options are :expect_failures [Boolean] doesnt return an exit code of non-zero if the command failed.
+  # @param opts [Hash] Alters the behaviour of the command. Valid options are  
+  #  :expect_failures [Boolean] doesnt return an exit code of non-zero if the command failed.  
+  #  :inventory_file [String] path to the inventory file to use with the task.
   # @return [Object] A result object from the task.The values available are stdout, stderr and result.
+  # rubocop:enable Layout/TrailingWhitespace
   def run_bolt_task(task_name, params = {}, opts = {})
     config_data = { 'modulepath' => File.join(Dir.pwd, 'spec', 'fixtures', 'modules') }
     target_node_name = targeting_localhost? ? 'litmus_localhost' : ENV['TARGET_HOST']
-    inventory_hash = File.exist?('inventory.yaml') ? inventory_hash_from_inventory_file : localhost_inventory_hash
+    inventory_hash = File.exist?('inventory.yaml') ? inventory_hash_from_inventory_file(opts[:inventory_file]) : localhost_inventory_hash
```

Comment:
Shouldn't the exist check be against `opts[:inventory_file]`? Otherwise, won't it still share as long as `inventory.yaml` exists (which it likely will if folks have used litmus before in their project)?

Reply:
We do _if_ we want to be able to mock features for localhost testing, as far as I can tell.

Reply:
Uggh, more is needed. 
the logic now is good. 
we also want to add the ability to add features to the inventory file at provision time. :) this will mitigate a lot of issues